content/2017-09-19-python-on-os-x.md

```diff
@@ -0,0 +1,204 @@
+title: Installing Python on OS X
+description: The Hows and Whys of Python versions
+date: 2017-09-19
+author: Danny Hermes (dhermes@bossylobster.com)
+tags: Programming, Mac OS X, Python, pyenc
+slug: python-on-os-x
+comments: true
+use_twitter_card: true
+use_open_graph: true
+use_schema_org: true
+twitter_site: @bossylobster
+twitter_creator: @bossylobster
+social_image: images/python-logo.png
+
+> **Caveat**: I stopped using a Mac for my development machine way
+> back in the fall of 2014. You probably shouldn't listen to me.
+
+### What Not to Do
+
+The first rule of using Python on any operating system: **don't use
+system Python**. The OS **relies** on that Python, including the packages
+(and package versions) it has installed. This means if it is modified in a
+non-official way by you, then the OS may not have something it depends on.
+
+For Mac OS X, system Python is installed in
+
+```
+/System/Library/Frameworks/Python.framework/Versions/2.7/bin/python
+```
+
+(YMMV, this is on a "current"-ish OS X 10.11.6). You couldn't edit it if
+you wanted to &mdash; not even with `sudo` &mdash; it's [restricted][8].
+
+Instead, most choose one of
+
+* the [pyenv][3] version manager
+* a custom Python from [Homebrew][4]
+* the [conda][7] package management system
+* "official" Python directly from [Python.org][6]
```

Comment:
@jonparrott Do you think it's worth putting this in some kind of order and then mentioning which ones are most prevalent? ISTM `brew` is by far the most prevalent, followed by "official", then `conda`, then `pyenv`, but I could be way wrong.

Also, is it worth noting that 

1. Homebrew will just do "exactly" what `pyenv` does: build Python from source. But it has a much smaller selection of Python versions.
1. `pyenv` can be installed via Homebrew (but I don't think it should, since `git clone https://github.com/pyenv/pyenv.git ~/.pyenv` is so smooth)
1. `conda` can be installed with `pyenv`

Reply:
Hrm.. no idea. Not even sure what the prevalence is.